src/simulator/diff_drive_model.cpp

```diff
@@ -0,0 +1,188 @@
+#include <nav_msgs/Odometry.h>
+#include "diff_drive_model.h"
+#include "shared/util/timer.h"
+
+using std::vector;
+using std::string;
+
+namespace diffdrive {
+
+CONFIG_BOOL(invert_x, "invert_linear_vel_cmds");
+CONFIG_BOOL(invert_z, "invert_angular_vel_cmds");
+CONFIG_FLOAT(linear_pos_accel_limit, "linear_pos_accel_limit");
+CONFIG_FLOAT(linear_neg_accel_limit, "linear_neg_accel_limit");
+CONFIG_FLOAT(angular_pos_accel_limit, "angular_pos_accel_limit");
+CONFIG_FLOAT(angular_neg_accel_limit, "angular_neg_accel_limit");
+CONFIG_FLOAT(max_angular_vel, "max_angular");
+CONFIG_FLOAT(max_linear_vel, "max_linear_vel");
+CONFIG_FLOAT(linear_odom_scale, "linear_odom_scale");
+CONFIG_FLOAT(angular_odom_scale, "angular_odom_scale");
+
+CONFIG_STRING(drive_topic, "drive_callback_topic");
+CONFIG_STRING(odom_topic, "diff_drive_odom_topic");
+
+DiffDriveModel::DiffDriveModel(const vector<string>& config_files, ros::NodeHandle* n) :
+	RobotModel(),
+	last_cmd_(),
+	t_last_cmd_(0),
+	angular_error_(0, 1),
+	config_reader_(config_files){
+        
+	drive_subscriber_ = n->subscribe(
+      CONFIG_drive_topic,
+      1,
+      &DiffDriveModel::DriveCallback,
+      this);
+	odom_publisher_ = n->advertise<nav_msgs::Odometry>(CONFIG_odom_topic, 1);
+    odometry_x = 0.0;
+    odometry_y = 0.0;
+    odometry_w = 0.0;
+    target_linear_vel = 0.0;
+    target_angular_vel = 0.0;
+    pose_.angle = 0;
+    odom_msg.header.frame_id = "/odom";
+    odom_msg.child_frame_id = "/base_link";
+
+}
+
+void DiffDriveModel::Step(const double &dt) {
+    ros::Time current_time = ros::Time::now();
+    // Update the linear velocity based on the linear acceleration limits
+	if (linear_vel < target_linear_vel) {
+		// Must increase linear speed
+		if (CONFIG_linear_pos_accel_limit == 0.0 || target_linear_vel - linear_vel < CONFIG_linear_pos_accel_limit)
+			linear_vel = target_linear_vel;
+		else
+			 linear_vel += CONFIG_linear_pos_accel_limit; 
+	} else if (linear_vel > target_linear_vel) {
+		// Must decrease linear speed
+		if (CONFIG_linear_neg_accel_limit == 0.0 || linear_vel - target_linear_vel < CONFIG_linear_neg_accel_limit)
+			linear_vel = target_linear_vel;
+		else
+			 linear_vel -= CONFIG_linear_neg_accel_limit; 
+	}
+
+	// Update the angular velocity based on the angular acceleration limits
+	if (angular_vel < target_angular_vel) {
+		// Must increase angular speed
+		if (CONFIG_angular_pos_accel_limit == 0.0
+			|| target_angular_vel - angular_vel < CONFIG_angular_pos_accel_limit)
+			angular_vel = target_angular_vel;
+		else
+			 angular_vel += CONFIG_angular_pos_accel_limit; 
+	} else if (angular_vel > target_angular_vel) {
+		// Must decrease angular speed
+		if (CONFIG_angular_neg_accel_limit == 0.0 
+			|| angular_vel - target_angular_vel < CONFIG_angular_neg_accel_limit)
+			angular_vel = target_angular_vel;
+		else
+			 angular_vel -= CONFIG_angular_neg_accel_limit; 
+	}
+
+    float forward_displacement = dt * linear_vel * CONFIG_linear_odom_scale;
```

Comment:
If a value is constant, please make the variable const.

Reply:
What value are you referring to? Nothing there is constant.

Reply:
`forward_displacement` is declared and then never modified in the code as far as I can tell. It isn't constant in that it will have the same value every time this function is called, but in that it will  only ever have one value each time it is called. So it should be constant. 

This is also true of several other variables in this functions scope and elsewhere.